connectors/serialization/src/main/java/serialization/ComplexTypeColumn.java

```diff
@@ -0,0 +1,256 @@
+                break;
+            }
+        }
+        if (allNulls && !existed.getType().equals(newColumn.getType())) {
+            insertNull(newColumn, existed.length - 1);
+            for (int i = 0; i < data.length; i++) {
+                if (data[i] == existed) {
+                    data[i] = newColumn;
+                    return;
+                }
+            }
+        }
+        for (int i = 0; i < newColumn.length; i++) {
+            Object value = newColumn.get(i);
+            if (existed.getType().equals(Types.BOOL)) {
+                value = ((int) value) == 1;
+            }
+            if (i == 0) {
+                existed.set(existed.length - 1, value); // replace added null
+            } else {
+                existed.add(value);
+            }
+        }
+    }
+
+    private void ensureSpace(int extraLength) {
+        if (length + extraLength > data.length) {
+            Column[] newData = new Column[data.length * 2 + Math.max(0, length + extraLength - data.length * 2)];
+            System.arraycopy(data, 0, newData, 0, data.length);
+            data = newData;
+        }
+    }
+
+    private void insertNull(Column column, int nullCount) {
+        if (column == null) return;
+        try {
+            Field data ;
+            if (column.getType().equals(Types.BIG_INT)) {
+                data = column.getClass().getSuperclass().getDeclaredField(DATA_FIELD_NAME);
+            } else {
+                data = column.getClass().getDeclaredField(DATA_FIELD_NAME);
+            }
+            data.setAccessible(true);
+            Class<?> componentType = data.get(column).getClass().getComponentType();
+            Object array = Array.newInstance(componentType.isPrimitive()
+                    ? getWrapperClassForPrimitive(componentType) : componentType, column.length + nullCount);
+            for (int j = nullCount, i = 0; i < column.length; j++, i++) {
+                if (column.getType().equals(Types.BOOL)) {
+                    boolean value = ((Integer) column.get(i)) == 1;
+                    Array.set(array, j, value);
+                } else {
+                    Array.set(array, j, column.get(i));
+                }
+            }
+            data.setAccessible(false);
+            column.empty();
+            column.addAll((convertToObjectArray(array)));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException("Something went wrong when inserting null", e);
+        }
+    }
+
+    private Object[] convertToObjectArray(Object array) {
+        Class ofArray = array.getClass().getComponentType();
+        if (ofArray.isPrimitive()) {
+            List ar = new ArrayList();
+            int length = Array.getLength(array);
+            for (int i = 0; i < length; i++) {
+                ar.add(Array.get(array, i));
+            }
+            return ar.toArray();
+        }
+        else {
+            return (Object[]) array;
```

Comment:
you don't really need these extra braces after else, just write it as
```
else
  return (Object[]) array;
```

Reply:
It's incorrect style for java. https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used